apps/emqx_auth/src/emqx_authn/emqx_authn_api.erl

```diff
@@ -838,23 +847,37 @@ listener_authenticator_position(
         end
     ).
 
-authenticator_users(post, #{bindings := #{id := AuthenticatorID}, body := UserInfo}) ->
-    add_user(?GLOBAL, AuthenticatorID, UserInfo);
+authenticator_users(post, #{bindings := #{id := AuthenticatorID}, body := UserInfo} = Req) ->
+    ActorNamespace = emqx_dashboard:get_namespace(Req),
+    case UserInfo of
+        #{<<"namespace">> := UserNamespace} when
+            ActorNamespace /= ?global_ns andalso UserNamespace /= ActorNamespace
+        ->
+            ?FORBIDDEN(<<"User not authorized to operate on requested namespace">>);
```

Comment:
Why only post is protected from accessing by not authorised admins? Why are put and delete not checked?